modules/signals/src/signal-store-feature.ts

```diff
@@ -0,0 +1,224 @@
+import {
+  EmptyFeatureResult,
+  MergeFeatureResults,
+  SignalStoreFeature,
+  SignalStoreFeatureResult,
+} from './signal-store-models';
+
+export function signalStoreFeature<F1 extends SignalStoreFeatureResult>(
```

Comment:
Can these overload types be refactored with variadic tuple types?

Reply:
No, it's not possible. Every next feature needs the type of the previous one, similar to [RxJS `pipe`](https://github.com/ReactiveX/rxjs/blob/master/src/internal/util/pipe.ts#L4).